pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/segment/FlushThresholdUpdateManager.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.controller.helix.core.realtime.segment;
+
+import com.linkedin.pinot.common.config.TableConfig;
+import com.linkedin.pinot.common.utils.CommonConstants;
+import com.linkedin.pinot.core.realtime.impl.kafka.KafkaHighLevelStreamProviderConfig;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Manager which maintains the flush threshold update objects for each table
+ */
+public class FlushThresholdUpdateManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FlushThresholdUpdateManager.class);
+
+  private ConcurrentMap<String, FlushThresholdUpdater> _flushThresholdUpdaterMap = new ConcurrentHashMap<>();
+
+  /**
+   * Check table config for flush size.
+   *
+   * If flush size < 0, create a new DefaultFlushThresholdUpdater with default flush size
+   * If flush size > 0, create a new DefaultFlushThresholdUpdater with given flush size.
+   * If flush size == 0, create new SegmentSizeBasedFlushThresholdUpdater if not already created. Create only 1 per table, because we want to maintain tuning information for the table in the updater
+   * @param realtimeTableConfig
+   * @return
+   */
+  public FlushThresholdUpdater getFlushThresholdUpdater(TableConfig realtimeTableConfig) {
+    String tableName = realtimeTableConfig.getTableName();
+
+    int tableFlushSize = getLLCRealtimeTableFlushSize(realtimeTableConfig);
+    if (tableFlushSize < 0) {
+      tableFlushSize = KafkaHighLevelStreamProviderConfig.getDefaultMaxRealtimeRowsCount();
```

Comment:
Shouldn't you get it from KafkaLowLevelStreamProvviderConfig? I think it goes to high level config if low level isnot configured.

Reply:
sorry, my bad